src/Compilers/CSharp/Test/Syntax/Parsing/NullableParsingTests.cs

```diff
@@ -0,0 +1,620 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Roslyn.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public class NullableParsingTests : ParsingTests
+    {
+        public NullableParsingTests(ITestOutputHelper output) :
+            base(output)
+        {
+        }
+
+        [WorkItem(23272, "https://github.com/dotnet/roslyn/issues/23272")]
+        [Fact]
+        public void PartialAccessibilityAndNullableArray()
+        {
+            UsingTree(
+@"class C
+{
+    privat C[]? F;
```

Comment:
Is `privat` typo intentional?

Reply:
Yes, a typo is necessary to hit the bug. I've added a comment.